Sources/DistributedSystem/ChannelCompression.swift

```diff
@@ -67,23 +99,40 @@ final class ChannelCompressionHandshakeServer: ChannelInboundHandler, RemovableC
 final class ChannelCompressionHandshakeClient: ChannelInboundHandler, RemovableChannelHandler {
     typealias InboundIn = ByteBuffer
 
-    private let loggerBox: Box<Logger>
-    private let compression: UInt8
-    private let streamHandler: NIOCore.ChannelHandler
+    private let distributedSystem: DistributedSystem
+    private let channelHandler: ChannelHandler
 
-    init(_ loggerBox: Box<Logger>, _ compression: UInt8, _ streamHandler: NIOCore.ChannelHandler) {
-        self.loggerBox = loggerBox
-        self.compression = compression
-        self.streamHandler = streamHandler
+    init(_ distributedSystem: DistributedSystem, _ channelHandler: ChannelHandler) {
+        self.distributedSystem = distributedSystem
+        self.channelHandler = channelHandler
     }
 
     func channelActive(context: ChannelHandlerContext) {
-        var buffer = ByteBufferAllocator().buffer(capacity: MemoryLayout<UInt8>.size)
-        buffer.writeInteger(compression, as: UInt8.self)
-        context.writeAndFlush(NIOAny(buffer), promise: nil)
-        if compression != 0 {
-            _ = context.pipeline.addHandler(ChannelCompressionInboundHandler(loggerBox), position: .after(streamHandler))
-            _ = context.pipeline.addHandler(ChannelCompressionOutboundHandler(), position: .last)
+        let compressionMode = distributedSystem.compressionMode
+        switch compressionMode {
+        case .disabled:
+            var buffer = ByteBufferAllocator().buffer(capacity: MemoryLayout<UInt8>.size)
+            buffer.writeInteger(UInt8(0))
```

Comment:
Can these magic 0, 1 and 2 can be defined once?

Reply:
Fixed.